dlib/dnn/layers.h

```diff
@@ -1712,32 +1712,40 @@ namespace dlib
             unsigned long new_window_size;
         };
 
-        class visitor_bn_input_no_bias
+        class visitor_disable_input_bias
         {
         public:
 
             template <typename T>
-            void set_input_no_bias(T&) const
+            void disable_input_bias(T&) const
             {
                 // ignore other layer types
             }
 
             template <typename U, typename E>
-            void set_input_no_bias(add_layer<layer_norm_, U, E>& l)
+            void disable_input_bias(add_layer<layer_norm_, U, E>& l)
             {
                 disable_bias(l.subnet().layer_details());
                 set_bias_learning_rate_multiplier(l.subnet().layer_details(), 0);
                 set_bias_weight_decay_multiplier(l.subnet().layer_details(), 0);
             }
 
             template <layer_mode mode, typename U, typename E>
-            void set_input_no_bias(add_layer<bn_<mode>, U, E>& l)
+            void disable_input_bias(add_layer<bn_<mode>, U, E>& l)
             {
                 disable_bias(l.subnet().layer_details());
                 set_bias_learning_rate_multiplier(l.subnet().layer_details(), 0);
                 set_bias_weight_decay_multiplier(l.subnet().layer_details(), 0);
             }
 
+            template <layer_mode mode, size_t N, template <typename> class R, typename U, typename E>
+            void disable_input_bias(add_layer<bn_<mode>, repeat<N, R, U>, E>& l)
+            {
+                disable_bias(l.subnet().get_repeated_layer(0).layer_details());
+                set_bias_learning_rate_multiplier(l.subnet().get_repeated_layer(0).layer_details(), 0);
+                set_bias_weight_decay_multiplier(l.subnet().get_repeated_layer(0).layer_details(), 0);
+            }
+
```

Comment:
Not sure about this... as I said in #2216, it works for some cases, but not all...